src/main/java/org/dependencytrack/policy/cel/mapping/ComponentProjection.java

```diff
@@ -70,16 +68,18 @@ public class ComponentProjection {
 
     @MappedField(sqlColumnName = "BLAKE3")
     public String blake3;
-
     @MappedField(protoFieldName = "resolved_license", sqlColumnName = "LICENSE_ID")
     public Long resolvedLicenseId;
 
     @MappedField(protoFieldName = "license_name", sqlColumnName = "LICENSE")
     public String licenseName;
 
+    @MappedField(protoFieldName = "published_at", sqlColumnName = "PUBLISHED_AT")
+    public Date publishedAt;
+
+    @MappedField(protoFieldName = "latest_version", sqlColumnName = "LATEST_VERSION")
+    public String latestVersion;
```

Comment:
These two fields should not have `@MappedField` annotations. It's a admittedly confusing limitation of the mapping works right now...

Essentially, fields annotated with `@MappedField` must be present in both the `Component` Protobuf type (referred to by `protoFieldName`), *and* the `COMPONENT` database table (referred to by `sqlColumnName`). The mapping does currently not work when `sqlColumnName` refers to a column that was `JOIN`ed.

The workaround is to keep the fields without annotation:

https://github.com/DependencyTrack/hyades-apiserver/blob/dbcf47e8e620d9d10278101e841d78c4871b48a6/src/main/java/org/dependencytrack/policy/cel/mapping/ComponentProjection.java#L82-L84

And then add the `JOIN`ed column by hand when the Protobuf field is a requirement:

https://github.com/DependencyTrack/hyades-apiserver/blob/dbcf47e8e620d9d10278101e841d78c4871b48a6/src/main/java/org/dependencytrack/policy/cel/CelPolicyQueryManager.java#L147-L151